code/game/objects/items/weapons/melee/misc.dm

```diff
@@ -106,3 +106,196 @@
 			bug.death(TRUE)
 		if(!QDELETED(M))
 			qdel(M)
+
+/obj/item/melee/spellblade
+	name = "spellblade"
+	desc = "An enchanted blade with a series of a runes along the side."
+	icon = 'icons/obj/guns/magic.dmi'
+	icon_state = "spellblade"
+	item_state = "spellblade"
+	hitsound = 'sound/weapons/rapierhit.ogg'
+	force = 25
+	armour_penetration = 50
+	block_chance = 50
+	///enchantment holder, gives it unique on hit effects.
+	var/datum/enchantment/enchant = null
+	///the cooldown and power of enchantments are multiplied by this var when its applied
+	var/power = 1
+
+/obj/item/melee/spellblade/Destroy()
+	QDEL_NULL(enchant)
+	return ..()
+
+/obj/item/melee/spellblade/afterattack(atom/target, mob/user, proximity, params)
+	. = ..()
+	enchant?.on_hit(target, user, proximity, src)
+
+/obj/item/melee/spellblade/attack_self(mob/user)
+	if(enchant)
+		return
+
+	var/list/options = list("Lightning",/// todo add icons for these
+							"Fire",
+							"Bluespace",
+							"Forcewall",)
+	var/list/options_to_type = list("Lightning" = /datum/enchantment/lightning,
+									"Fire" = /datum/enchantment/fire,
+									"Bluespace" = /datum/enchantment/bluespace,
+									"Forcewall" = /datum/enchantment/forcewall,)
+
+	var/choice = show_radial_menu(user, src, options)
+	if(!choice)
+		return
+	add_enchantment(options_to_type[choice], user)
+
+/obj/item/melee/spellblade/proc/add_enchantment(new_enchant, mob/living/user, intentional = TRUE)
+	var/datum/enchantment/E = new new_enchant
```

Comment:
You can do without the variable `E` here. Just as a note. Don't have to remove it if you think it's cleaner